ui/src/main.rs

```diff
@@ -361,6 +368,57 @@ fn app(cx: Scope) -> Element {
     let desktop = use_window(cx);
     let state = use_shared_state::<State>(cx)?;
 
+    // Automatically select the best implementation for your platform.
+    let inner = state.inner();
```

Comment:
i recommend moving this and the use_future to the end of this function. the function was laid out so the `use_future`s were declared after `main_element`, which allowed a dev to see the end result of this function without having to scroll past all the `use_future`s